app.js

```diff
@@ -0,0 +1,198 @@
+let express = require('express');
+let User = require('./models/user');
+let models = require('./models/index');
+
+let auth = require('./lib/auth');
+let logger = require('./lib/logger');
+let metrics = require('./lib/metrics');
+
+let app = express();
+let server = app.listen(8080, () => logger.log('app', 'listening on 8080'));
+
+if (process.env.LOCAL) {
+	models.initForLocal();
+	models.dataReset();
+}
+
+app.use(express.json());
+
+// access log
+app.use((req, res, next) => {
+	logger.log('router', 'incoming: ' + req.url);
+	next();
+});
+
+// convenience for testing
+app.post('/exit', (req, res) => {
+	res.end();
+	if (models.local) {
+		server.close();
+	}
+});
+
+// root
+app.get('/', (req, res) => {
+	let start = Date.now();
+	metrics.increment('req_get_root');
+	res.json({status: 'ok'});
+	let duration = Date.now() - start;
+	metrics.record('lat_index', duration);
+});
+
+// auth
+app.use((req, res, next) => {
+	let start = Date.now();
+	metrics.increment('auth');
+	let accessToken = req.headers.authorization;
+	let isAuthenticated = auth.authenticate(accessToken);
+	if (!isAuthenticated) {
+		metrics.increment('unauthorized');
+		res.status(401).json({error: 'unauthorized'});
+		let duration = Date.now() - start;
+		metrics.record('lat_auth_unauthorized', duration);
+		return;
+	}
+
+	let duration = Date.now() - start;
+	metrics.record('lat_auth_authorized', duration);
+	next();
+});
+
+app.get('/v1/users', async (req, res) => {
+	let start = Date.now();
```

Comment:
I don't really like the boilerplate-to-actual-functionality ratio. While I understand in some cases it's important to measure in-handler metrics you could get away with just writing simple middleware which would do that for you automatically and without cluttering the code.

Reply:
@sf-r if you were going to approach this, how would you do it?

Reply:
agreed, you're right @tmarciszewski 

think we can get around this by utilizing `next()` in our existing mws and sandwich them with two `use` mws that make use of res.local 👍 

thoughts?

Reply:
Yeah. That's probably what I'd do. You have full control over order of middleware execution so it's quite easy to move the most repeatable code to two middlewares which would execute at the start and at the end of request.